src/ansys/units/dimensions.py

```diff
@@ -1,142 +1,106 @@
-"""Provides the ``Dimension`` class."""
-import ansys.units as ansunits
-from ansys.units.utils import condense, filter_unit_term
+"""Provides the ``Dimensions`` and ``BaseDimensions`` class."""
+from enum import Enum
 
 
-class Dimensions(object):
+class BaseDimensions(Enum):
     """
-    Initializes a ``Dimensions`` object using a dimensions list or a unit string.
+    Supplies all valid base dimensions used in dimensional analysis.
+
+    Used as dictionary keys for defining a `Dimensions` object.
+    """
+
+    MASS = 0
+    LENGTH = 1
+    TIME = 2
+    TEMPERATURE = 3
+    TEMPERATURE_DIFFERENCE = 4
+    ANGLE = 5
+    CHEMICAL_AMOUNT = 6
+    LIGHT = 7
+    CURRENT = 8
+    SOLID_ANGLE = 9
+
+
+class Dimensions:
+    """
+    Initializes a ``Dimensions`` object from a dictionary of ``BaseDimensions`` and
+    power.
 
     Parameters
     ----------
-    units : str, None
-        Unit string of the quantity.
-    dimensions : list, None
-        List of the dimensions.
-    unit_sys : str, None
-        Unit system for creating units.
+    dimensions_container : dictionary, optional
+        Dictionary of {``BaseDimensions``: power, ...}.
 
     Returns
     -------
     Dimensions instance.
+        On which mathematical dimensional analysis is performed.
+        Default is dimensionless.
     """
```

Comment:
This is positioned as a class docstring - it should be moved inside `__init__`, but also note that `__init__` just returns None..

Reply:
I can make these changes. Can I get some more info on why these should be in the init and returns None?

Reply:
Yeah, good point, nothing is returned technically.  The object is constructed.

Reply:
Should this be changed on Unit, UnitRegistry, UnitSystem and Quantity also?

Reply:
If they are similar to this, yes.